pkg/storage/watch_cache.go

```diff
@@ -250,6 +250,20 @@ func (w *watchCache) GetAllEventsSinceThreadUnsafe(resourceVersion uint64) ([]wa
 	if size > 0 {
 		oldest = w.cache[w.startIndex%w.capacity].resourceVersion
 	}
+	if resourceVersion == 0 {
+		// resourceVersion = 0 means that we don't require any specific starting point
+		// and we would like to start watching from ~now.
+		// However, to keep backward compatibility, we additionally need to return the
+		// current state and only then start watching from that point.
+		//
+		// TODO: In v2 api, we should stop returning the current state - #13969.
+		allItems := w.store.List()
```

Comment:
Does this list go out to hit etcd? I guess it doesn't matter since this feature can't be used safely anyway, but if it doesn't go out to hit etcd there's no guarantee the combined state returned by the list would be a complete/correct one as of any particular resource version.